mdm/enroll/service.go

```diff
@@ -167,13 +167,21 @@ func (svc *service) findOrMakeMobileconfig(id string, f interface{}) (profile.Mo
 	return p.Mobileconfig, nil
 }
 
-func (svc *service) Enroll(ctx context.Context) (profile.Mobileconfig, error) {
-	return svc.findOrMakeMobileconfig(EnrollmentProfileId, svc.MakeEnrollmentProfile)
+func (svc *service) Enroll(ctx context.Context, id string) (profile.Mobileconfig, error) {
+	if id == "" {
+		return svc.findOrMakeMobileconfig(EnrollmentProfileId, svc.MakeEnrollmentProfile)
```

Comment:
This does not work. When MakeEnrollmentProfile takes an id argument, the findOrMakeMobileconfig method fails. I know what the problem is, but I'm not sure what the best way to add the id to the checkin URL with minimal code changes. Do you have any suggestions, @groob ?